packages/aws-cdk-lib/aws-kinesis/lib/resource-policy.ts

```diff
@@ -0,0 +1,53 @@
+import { Construct } from 'constructs';
+import { CfnResourcePolicy } from './kinesis.generated';
+import { IStream } from './stream';
+import { PolicyDocument } from '../../aws-iam';
+import { Resource } from '../../core';
+
+/**
+ * Properties to associate a data stream with a policy
+ */
+export interface ResourcePolicyProps {
+  /**
+   * The stream this policy applies to.
+   */
+  readonly stream: IStream;
+
+  /**
+   * IAM policy document to apply to a data stream.
+   * @default - empty policy document
+   */
+  readonly policyDocument?: PolicyDocument;
+}
+
+/**
+ * The policy for a data stream or registered consumer
+ *
+ * Policies define the operations that are allowed on this resource.
+ *
+ * You almost never need to define this construct directly.
+ *
+ * All AWS resources that support resource policies have a method called
+ * `addToResourcePolicy()`, which will automatically create a new resource
+ * policy if one doesn't exist yet, otherwise it will add to the existing
+ * policy.
+ *
+ * Prefer to use `addToResourcePolicy()` instead.
+ */
+export class ResourcePolicy extends Resource {
```

Comment:
It seems that there's some inconsistency in naming this class across the CDK. In some cases it is just called `ResourcePolicy` but in others it is named `<resourceName>Policy` i.e. `BuckeyPolicy`, `TopicPolicy`. I feel like I prefer the convention of using the more specific name, so in this case `StreamPolicy` but it's not something to block on. Curious to hear your thoughts on this.

Reply:
The name "ResourcePolicy" is aligned with [CFN](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-kinesis-resourcepolicy.html) resource naming conventions.

For reference, "TopicPolicy" is also "TopicPolicy" in [CFN](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-sns-topicpolicy.html).

I considered using "StreamPolicy", but avoided it for now since it would be inconsistent with CFN.
I'm honestly unsure about this - which naming convention do you think would be better?

Reply:
I see, in this case then let's just keep it consistent with the CFN resource name.